python/triton/language/extra/hip/libdevice.py

```diff
@@ -0,0 +1,438 @@
+from triton.language import core
+
+
+@core.builtin
+@core._tensor_member_fn
+def abs(x, _builder=None):
+    x = core._to_tensor(x, _builder)
+    if x.dtype.is_floating():
+        return core.tensor(_builder.create_fabs(x.handle), x.type)
+    elif x.dtype.is_int_signed():
+        return core.tensor(_builder.create_iabs(x.handle), x.type)
+    else:
+        assert False, f"Unexpected dtype {x.dtype}"
+
+
+@core.builtin
+@core._tensor_member_fn
+def floor(x, _builder=None):
+    x = core._to_tensor(x, _builder)
+    if x.dtype.is_floating():
+      return core.tensor(_builder.create_floor(x.handle), x.type)
+    else:
```

Comment:
why not call ocml functions? Though they are just wrappers of llvm intrinsics.

Reply:
The required "ocml functions" also boiled down to llvm intrinsics. I modified this part of the code.